test/aievec/conv2d_f32_unaligned_loads.mlir

```diff
@@ -0,0 +1,100 @@
+// This test case will directly return the result generated from -affine-super-vectorize
+// because in transfer_read %arg2[%arg3, %arg4], dim 1's memref shape size(2046) cannot be
+// divisible by the vector lanes(8).
+
+// RUN: aie-opt %s --affine-loop-unroll="unroll-full unroll-full-threshold=3" --canonicalize -affine-super-vectorize="virtual-vector-size=8" --aie-vectorize -split-input-file -unaligned-loads-check=true --debug-only=aie-vect 2>&1 | FileCheck %s
+
+func.func @conv2d (%A: memref<2048x2048xf32>, %B: memref<9xf32>, %C: memref<2046x2046xf32>) {
+    affine.for %arg3 = 0 to 2046 {
+        affine.for %arg4 = 0 to 2046 {
+            //3x3 stencil
+            affine.for %arg5 = 0 to 3 {
+                affine.for %arg6 = 0 to 3 {
+                    //Load the output point
+                    %ci = affine.load %C[%arg3, %arg4] : memref<2046x2046xf32>
+                     %a11 = affine.load %A[%arg3+%arg5, %arg4+%arg6] : memref<2048x2048xf32>
+                     %b11 = affine.load %B[3*%arg5+%arg6] : memref<9xf32>
+                     %p11 = arith.mulf %a11, %b11 : f32
+                     %c11 = arith.addf %ci, %p11 : f32
+                     //Store accumulated sum
+                     affine.store %c11, %C[%arg3, %arg4] : memref<2046x2046xf32>
+                }
+            }
+        }
+    }
+    return
+}
+//CHECK: %0 = vector.transfer_read %arg2[%arg3, %arg4], %cst {in_bounds = [true]} : memref<2046x2046xf32>, vector<8xf32>'s dim 1's shape size cannot be divisible by the vector lanes.
```

Comment:
usually we put the check lines first, consistent with llvm/utils/update_mlir_test_checks.py